@here/harp-olp-utils/lib/OlpDataProvider.ts

```diff
@@ -93,33 +89,43 @@ export class OlpDataProvider implements DataProvider {
      * @param abortSignal Optional AbortSignal to cancel the request.
      * @returns A promise delivering the data as an [[ArrayBufferLike]], or any object.
      */
-    getTile(tileKey: TileKey, abortSignal?: AbortSignal): Promise<ArrayBufferLike | {}> {
+    async getTile(tileKey: TileKey, abortSignal?: AbortSignal): Promise<ArrayBufferLike | {}> {
         if (this.m_versionLayerClient === undefined) {
             throw new Error("OlpDataProvider is not connected.");
         }
-        return this.m_versionLayerClient
-            .getData(
+
+        try {
+            const response = await this.m_versionLayerClient.getData(
                 new DataRequest().withQuadKey(tileKey).withVersion(this.m_catalogVersion),
                 abortSignal
-            )
-            .then(response => {
-                if (response.status !== 200) {
-                    throw new Error(response.statusText);
-                }
-                return response.arrayBuffer();
-            })
-            .catch(error => {
-                // 204 - NO CONTENT, no data exists at the given tile.
-                if (error.name === "HttpError" && error.status === 204) {
-                    return {};
-                }
-
-                logger.error(
-                    `Error loading tile ${tileKey.mortonCode()} for catalog ${
-                        this.params.hrn
-                    }: ${error}`
-                );
+            );
+            if (abortSignal && abortSignal.aborted) {
+                // Safety belt if `getData` doesn't really support abort signal.
+                const err = new Error("Aborted");
+                err.name = "AbortError";
+                throw err;
+            }
+            if (response.status !== 200) {
+                throw new Error(response.statusText);
+            }
+            return response.arrayBuffer();
+        } catch (error) {
+            if (error.name === "AbortError" || error.message === "AbortError: Aborted") {
+                // Rethrow abort errors as they shall be handled on higher level.
+                throw error;
```

Comment:
do you need to rethrow the abort error? why not take the if(abortSignal....) code above out of the try / catch?

The same for the response as well, or do I miss something?

Reply:
AbortError may also be thrown by proper implementation of `getData)_` or `arrayBuffer()`that just uses system fetch API.

AbortError should be rethrown because we cannot even handle it and it's not a point when we can decide what to do with this error.

> why not take the if(abortSignal....) code above out of the try / catch?

It can be moved ... but then it doesn't make sense. The real reason for this, is outlined in comment - if `getTile` ignored signal for whatever reason, just handle `abortSignal` API by ourselves and skip call to `arrayBuffer`.

Reply:
And this whole discussion is here only because someone decided to effectively add "catch(...) { log(...); return {}" only not to have duplicate log messages for tile load error (TileLoader would also log if we would rethrow).

So, good catch is no catch or catch that _rethrows_ exceptions it doesn't handle explicitly
Proper catch in this case should be
 - in this case, we handle explcitly HTTPError / 204
 - rethrow everything

But as i find value in verbosity of this logging message (includes catalog nrn), i left it as is with excplicit support for rethrowing `AbortError`s.

Reply:
Opps, I commented on the wrong line, I mean why do we have the rethrow on line 106, but I guess that it is because the getData implementation as stated may not support it.

Good point about the skip to arrayBuffer, I thought it would throw an error, and so you would automatically skip it, but it doesn't.